pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -311,12 +311,13 @@ func (k *Kaniko) compileJobSpec(namespace string,
 							},
 						},
 					},
-					RestartPolicy:     v1.RestartPolicyNever,
-					NodeSelector:      buildOptions.NodeSelector,
-					NodeName:          buildOptions.NodeName,
-					Affinity:          buildOptions.Affinity,
-					PriorityClassName: buildOptions.PriorityClassName,
-					Tolerations:       buildOptions.Tolerations,
+					RestartPolicy:      v1.RestartPolicyNever,
+					NodeSelector:       buildOptions.NodeSelector,
+					NodeName:           buildOptions.NodeName,
+					Affinity:           buildOptions.Affinity,
+					PriorityClassName:  buildOptions.PriorityClassName,
+					Tolerations:        buildOptions.Tolerations,
+					ServiceAccountName: buildOptions.ServiceAccount,
```

Comment:
```suggestion
					ServiceAccount: buildOptions.ServiceAccount,
```

Reply:
this attribute is of `v1.PodSpec`, can't change it. changed RHS instead.